identifier-forest/src/order/_order.rs

```diff
@@ -0,0 +1,588 @@
+//! This file defines how an array of four identifiers / an identifier quad can
+//! be ordered
+
+use crate::order::Position;
+use crate::Identifier;
+
+use std::cmp::Ordering;
+use std::marker::PhantomData;
+
+/// Number of different positions to specify in a [`FixedOrder4`].
+///
+/// It correspond to the number of elements in an identifier quad (4).
+pub const ORDER4_NB_OF_TERMS: usize = 4;
```

Comment:
Is this constant really useful? Since many parts of the code are *structurally* dependant it being 4 (e.g. the 4 parameters A, B, C, D of FixedOrder4) and since the number 4 occurs in many type or constant names in this module, I think that replacing this constant with "4" everywhere it occurs would make the code more readable, without any loss in generality or maintainability.

Reply:
I removed the constant in 62bb9a1b89ccd86b757f5a81e1dbb2ce9929aeca 

It was mostly tentative to document the crate and help implementing a triple forest in the future but I agree that it ended up being code bloat.